services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/FlowConnectedDeviceSpec.groovy

```diff
@@ -637,16 +658,50 @@ srcLldpDevices=#newSrcEnabled, dstLldpDevices=#newDstEnabled"() {
             }
         }
 
-        then: "Device is also visible as a flow device"
+        and: "Device is also visible as a flow device"
+        with(northbound.getFlowConnectedDevices(flow.id)) {
+            it.source.lldp.size() == 1
+            verifyEquals(it.source.lldp.first(), lldpData)
+        }
+
+        and: "Device is NOT visible as a default flow device"
+        northbound.getFlowConnectedDevices(defaultFlow.id).source.lldp.empty
+
+        when: "Another device sends an lldp packet into a flow port with vlan different from flow vlan"
+        def lldpData2 = LldpData.buildRandom()
+        def vlan = flow.source.vlanId - 1
+        new ConnectedDevice(traffExamProvider.get(), tg, vlan).withCloseable {
+            it.sendLldp(lldpData2)
+        }
+
+        then: "Corresponding device is detected on a switch port with reference to corresponding default flow"
         Wrappers.wait(WAIT_OFFSET) {
-            with(northbound.getFlowConnectedDevices(flow.id)) {
-                it.source.lldp.size() == 1
-                verifyEquals(it.source.lldp.first(), lldpData)
+            with(northboundV2.getConnectedDevices(sw.dpId).ports) {
+                it.size() == 1
+                it[0].portNumber == tg.switchPort
+                it[0].lldp.size() == 2
+                it[0].lldp.last().flowId == defaultFlow.id
+                it[0].lldp.last().vlan == vlan
```

Comment:
1. the order of devices is not guarantee, so the device can be the first in some cases.  
2. There is one limitation for default flows - vlan for coming packets equals to 0. Even if you push packet with vlan 11 to default flow - you will get connected device with vlan 0.

Could you please add more tests for default rules? For example:
one switch flow:
inVlan: 0,  outVlan: 0
SeveralSwitchFlow:
inVlan: 0,  outVlan: 0
inVlan: 0,  outVlan: X

Reply:
1. Do you find it reasonable to sort devices by timeLastSeen? I would expect some ordering there

Reply:
2. So to clarify - I won't be able to see vlans neither per-switch nor per-flow if there is a default flow present?

Reply:
Added 
```
SeveralSwitchFlow:
inVlan: 0, outVlan: 0
inVlan: X, outVlan: 0
```
Other cases tested manually.